doc/en/backwards-compatibility.rst

```diff
@@ -3,6 +3,57 @@
 Backwards Compatibility Policy
 ==============================
 
+.. versionadded: 6.0
+
+pytest is actively being evolved and as a project that is been decades in the making,
+we keep learning about new and better structures to express different details about testing.
+
+While we implement those we try to ensure easy transition and don't want to impose unnecessary churn on our users and community/plugin authors.
+
+We are considering multipe Types of backward compatibility considering transitions
+
+a) trivial: api's which trivially translate to the new mechanism,
+   and do not cause problematic details.
+
+   We try to support those indefinitely while encouraging users to switch to newer/better mechanisms.
+
+b) transitional: the old and new api don't conflict
+   and we can help while supporting both for a prolonged time
+
+   We will only start the removal of deprecated functionality in major releases (e.g. if we deprecate something in 3.0 we will start to remove it in 4.0), and keep it around for at least two minor releases (e.g. if we deprecate something in 3.9 and 4.0 is the next release, we start to remove it in 5.0, not in 4.0).
+
+   When the deprecation expires (e.g. 4.0 is released), we won't remove the deprecated functionality immediately, but will use the standard warning filters to turn them into **errors** by default. This approach makes it explicit that removal is imminent, and still gives you time to turn the deprecated feature into a warning instead of an error so it can be dealt with in your own time. In the next minor release (e.g. 4.1), the feature will be effectively removed.
+
+
+
+c) true breakage: should only to be cosidered when a combination of transition seems unreasonably unsustainable and would offset important development/features by years
+   In addition they shuld be limited to apis where the number of actual users is very small and can be coordinated.
+
+   Examples for such upcinging changes:
+
+   * removal of ``pytest_runtest_protocol/nextitem``
```

Comment:
Here we should either 1) explain why those should go and which impact they have or 2) link to an issue number

I understand some of them might not have an issue number yet (I think this one in particular doesn't), but as it is the examples are meaningless to users (and I suspect even some core devs), so they are not very useful.